==890== Memcheck, a memory error detector
==890== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==890== Using Valgrind-3.16.1 and LibVEX; rerun with -h for copyright info
==890== Command: ./piMsg4
==890== Parent PID: 503
==890== 
==890== Thread 4:
==890== Invalid read of size 8
==890==    at 0x10A134: packet_to_binary (piMsg4.c:270)
==890==    by 0x10A09F: qsend (piMsg4.c:173)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890==  Address 0x4a44988 is 8 bytes inside a block of size 16 free'd
==890==    at 0x484AFE0: free (vg_replace_malloc.c:538)
==890==    by 0x10A03F: main (piMsg4.c:140)
==890==  Block was alloc'd at
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x109F3F: main (piMsg4.c:116)
==890== 
==890== Invalid read of size 1
==890==    at 0x10A1AC: packet_to_binary (piMsg4.c:280)
==890==    by 0x10A09F: qsend (piMsg4.c:173)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890==  Address 0x4a44980 is 0 bytes inside a block of size 16 free'd
==890==    at 0x484AFE0: free (vg_replace_malloc.c:538)
==890==    by 0x10A03F: main (piMsg4.c:140)
==890==  Block was alloc'd at
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x109F3F: main (piMsg4.c:116)
==890== 
==890== Invalid read of size 1
==890==    at 0x10A1F0: packet_to_binary (piMsg4.c:281)
==890==    by 0x10A09F: qsend (piMsg4.c:173)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890==  Address 0x4a44981 is 1 bytes inside a block of size 16 free'd
==890==    at 0x484AFE0: free (vg_replace_malloc.c:538)
==890==    by 0x10A03F: main (piMsg4.c:140)
==890==  Block was alloc'd at
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x109F3F: main (piMsg4.c:116)
==890== 
==890== Invalid read of size 8
==890==    at 0x10A298: packet_to_binary (piMsg4.c:290)
==890==    by 0x10A09F: qsend (piMsg4.c:173)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890==  Address 0x4a44988 is 8 bytes inside a block of size 16 free'd
==890==    at 0x484AFE0: free (vg_replace_malloc.c:538)
==890==    by 0x10A03F: main (piMsg4.c:140)
==890==  Block was alloc'd at
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x109F3F: main (piMsg4.c:116)
==890== 
==890== Invalid free() / delete / delete[] / realloc()
==890==    at 0x484AFE0: free (vg_replace_malloc.c:538)
==890==    by 0x10A117: qsend (piMsg4.c:191)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890==  Address 0x4a44980 is 0 bytes inside a block of size 16 free'd
==890==    at 0x484AFE0: free (vg_replace_malloc.c:538)
==890==    by 0x10A03F: main (piMsg4.c:140)
==890==  Block was alloc'd at
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x109F3F: main (piMsg4.c:116)
==890== 
==890== 
==890== Process terminating with default action of signal 2 (SIGINT)
==890==    at 0x4986B54: __libc_read (read.c:26)
==890==    by 0x4986B54: read (read.c:24)
==890==    by 0x4933577: _IO_file_underflow@@GLIBC_2.17 (fileops.c:517)
==890==    by 0x49346AF: _IO_default_uflow (genops.c:362)
==890==    by 0x492E98B: getchar (getchar.c:39)
==890==    by 0x109EDB: main (piMsg4.c:105)
==890== 
==890== HEAP SUMMARY:
==890==     in use at exit: 7,360 bytes in 9 blocks
==890==   total heap usage: 105 allocs, 97 frees, 51,384 bytes allocated
==890== 
==890== Thread 1:
==890== 8 bytes in 1 blocks are still reachable in loss record 1 of 9
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x486B6FF: start_thread (in /usr/lib/libpigpiod_if2.so.1)
==890==    by 0x486D95F: pigpio_start (in /usr/lib/libpigpiod_if2.so.1)
==890==    by 0x109E43: main (piMsg4.c:86)
==890== 
==890== 11 bytes in 1 blocks are definitely lost in loss record 2 of 9
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x493FE5F: strdup (strdup.c:42)
==890==    by 0x109FB3: main (piMsg4.c:127)
==890== 
==890== 13 bytes in 1 blocks are definitely lost in loss record 3 of 9
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x109DB3: enqueue (queue.h:121)
==890==    by 0x10A01B: main (piMsg4.c:137)
==890== 
==890== 56 bytes in 1 blocks are still reachable in loss record 4 of 9
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x48749D3: callback (in /usr/lib/libpigpiod_if2.so.1)
==890==    by 0x10A94B: receiveHandshake (piMsg4.c:468)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890== 
==890== 56 bytes in 1 blocks are still reachable in loss record 5 of 9
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x48748F3: callback (in /usr/lib/libpigpiod_if2.so.1)
==890==    by 0x10A96F: receiveHandshake (piMsg4.c:469)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890== 
==890== 272 bytes in 1 blocks are possibly lost in loss record 6 of 9
==890==    at 0x484C164: calloc (vg_replace_malloc.c:760)
==890==    by 0x40118CB: allocate_dtv (dl-tls.c:348)
==890==    by 0x40118CB: _dl_allocate_tls (dl-tls.c:594)
==890==    by 0x489A047: allocate_stack (allocatestack.c:622)
==890==    by 0x489A047: pthread_create@@GLIBC_2.17 (pthread_create.c:660)
==890==    by 0x486B73B: start_thread (in /usr/lib/libpigpiod_if2.so.1)
==890==    by 0x486D95F: pigpio_start (in /usr/lib/libpigpiod_if2.so.1)
==890==    by 0x109E43: main (piMsg4.c:86)
==890== 
==890== 272 bytes in 1 blocks are possibly lost in loss record 7 of 9
==890==    at 0x484C164: calloc (vg_replace_malloc.c:760)
==890==    by 0x40118CB: allocate_dtv (dl-tls.c:348)
==890==    by 0x40118CB: _dl_allocate_tls (dl-tls.c:594)
==890==    by 0x489A047: allocate_stack (allocatestack.c:622)
==890==    by 0x489A047: pthread_create@@GLIBC_2.17 (pthread_create.c:660)
==890==    by 0x109E7B: main (piMsg4.c:90)
==890== 
==890== 272 bytes in 1 blocks are possibly lost in loss record 8 of 9
==890==    at 0x484C164: calloc (vg_replace_malloc.c:760)
==890==    by 0x40118CB: allocate_dtv (dl-tls.c:348)
==890==    by 0x40118CB: _dl_allocate_tls (dl-tls.c:594)
==890==    by 0x489A047: allocate_stack (allocatestack.c:622)
==890==    by 0x489A047: pthread_create@@GLIBC_2.17 (pthread_create.c:660)
==890==    by 0x109E97: main (piMsg4.c:93)
==890== 
==890== 6,400 bytes in 1 blocks are still reachable in loss record 9 of 9
==890==    at 0x4849E4C: malloc (vg_replace_malloc.c:307)
==890==    by 0x10A977: receiveHandshake (piMsg4.c:471)
==890==    by 0x4899647: start_thread (pthread_create.c:477)
==890==    by 0x4994C9B: thread_start (clone.S:78)
==890== 
==890== LEAK SUMMARY:
==890==    definitely lost: 24 bytes in 2 blocks
==890==    indirectly lost: 0 bytes in 0 blocks
==890==      possibly lost: 816 bytes in 3 blocks
==890==    still reachable: 6,520 bytes in 4 blocks
==890==         suppressed: 0 bytes in 0 blocks
==890== 
==890== For lists of detected and suppressed errors, rerun with: -s
==890== ERROR SUMMARY: 16 errors from 10 contexts (suppressed: 0 from 0)
